e2e/trust/sign_test.go

```diff
@@ -0,0 +1,90 @@
+package trust
+
+import (
+	"os"
+	"fmt"
+	"testing"
+
+	"github.com/gotestyourself/gotestyourself/fs"
+	"github.com/gotestyourself/gotestyourself/icmd"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	notaryURL    = "https://notary-server:4443"
+	alpineImage  = "registry:5000/alpine:3.6"
+	alpineSha    = "641b95ddb2ea9dc2af1a0113b6b348ebc20872ba615204fbe12148e98fd6f23d"
+	busyboxImage = "registry:5000/busybox:1.27.2"
+	busyboxSha   = "030fcb92e1487b18c974784dcc110a93147c9fc402188370fbfd17efabffc6af"
+	localImage   = "registry:5000/signlocal:v1"
+	signImage    = "registry:5000/sign:v1"
+)
+
+func TestSignWithTrustSign(t *testing.T) {
+	dir := setupConfigFile(t)
+	defer removeConfigFile(dir)
+	icmd.RunCmd(icmd.Command("docker", "pull", alpineImage)).Assert(t, icmd.Success)
+	icmd.RunCommand("docker", "tag", alpineImage, signImage).Assert(t, icmd.Success)
+	result := icmd.RunCmd(
+		icmd.Command("docker", "trust", "sign", signImage),
+		withTrustAndPassphrase("root_password", "repo_password", dir))
+	result.Assert(t, icmd.Success)
+	assert.Contains(t, result.Stdout(), fmt.Sprintf("v1: digest: sha256:%s", alpineSha))
+
+}
+
+func TestSignWithTrustSignlocal(t *testing.T) {
```

Comment:
This case is not just `--local`, it's actually that it's writing over a previous signature, is that right? Should this be `TestSignWithLocalOverrides` or something like that?

Reply:
The only way of writing over the previous signature is with --local and that's really its only use

Reply:
Ok, so maybe `TestSignWithLocalFlag` ?